### PR TITLE
NFS storage concept

### PR DESCRIPTION
**Pull request by pbochynski**
**Description**

Changes proposed in this pull request:

- concept for NFS storage in Kyma runtime

### CODE REVIEW

**Comment by anishj0shi** [comment]
i quickly tested gardener's reconciliation behavior by scheduling an azure storage account in a VNet. and the standard reconciliation of clusters works without any errors. 
But deletion of cluster fails due to private endpoint created in the shoot subnet. 
```
deleting Subnet: (Name "shoot--ccloud-poc--ncm-poc-nodes" / Virtual Network Name "shoot--ccloud-poc--ncm-poc" / Resource Group "shoot--ccloud-poc--ncm-poc"):
 network.SubnetsClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="InUseSubnetCannotBeDeleted" 
Message="Subnet shoot--ccloud-poc--ncm-poc-nodes is in use by /subscris/ENDPOINT-STORAGE.NIC.<omitted>/ipConfigurations/PRIVATEENDPOINTIPCONFIG.<omitted> 
and cannot be deleted. 
In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet." Details=[]]
```
this creates a pre-requisite condition for us that, we'd need to clean up the storage account and the private endpoint in the shoot network.

**Comment by pbochynski** [comment]
> i quickly tested gardener's reconciliation behavior by scheduling an azure storage account in a VNet. and the standard reconciliation of clusters works without any errors. But deletion of cluster fails due to private endpoint created in the shoot subnet.
> this creates a pre-requisite condition for us that, we'd need to clean up the storage account and the private endpoint in the shoot network, before we delete the cluster. if we agree to this, then we can merge this PR.

Yes - we can assume that our controllers will listen to the GardenerCluster resource and we will delete all the resources created by our controllers. It is mentioned already in the document:
```
 storage instance can block cluster deprovisioning (not a big deal as storage controller in the KCP should clean it up anyway)
```

File: concepts/nfs-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+**Using shoot VPC**
+
+![](nfs-shoot-vpc.drawio.svg)
+
+**Pros:**
+- simpler implementation (provisioning)
+
+**Cons:**
+- if clusters are deleted and created with different shoot name we can face [issues](https://cloud.google.com/filestore/docs/create-instance-issues#system_limit_for_internal_resources_has_been_reached_error_when_creating_an_instance) with private network quota for Filestore (it is not a common usage pattern)
+- storage instance can block cluster deprovisioning (not a big deal as storage controller in the KCP should clean it up anyway)
```
Comment (anishj0shi) [inline concepts/nfs-storage/README.md:99]:
```suggestion
- storage instances and private endpoints in the shoot subnet can block cluster de-provisioning (not a big deal as storage controller in the KCP should clean them up anyway)
```

**Review by anishj0shi — APPROVED** [review]
/lgtm

**Pull request merged**